dist/Storable/Storable.pm

```diff
@@ -27,8 +27,9 @@ our @EXPORT_OK = qw(
 
 our ($canonical, $forgive_me);
 
+our $VERSION;
 BEGIN {
-  our $VERSION = '3.21';
```

Comment:
Why was this changed?

Reply:
Global variables should be defined outside of `BEGIN` block if used later.
```
> perl -E 'package Abc; use strict; use warnings; BEGIN { our $V = 42 } say $V '
Variable "$V" is not imported at -e line 1.
Global symbol "$V" requires explicit package name (did you forget to declare "my $V"?) at -e line 1.
```
Could you provide an example of a static version parsers broken with such a change?

Reply:
`$VERSION` is not used again in this file, there is no point to this change.

Module::Metadata is an example of a static version parser, there are others such as EUMM and PAUSE. They will use the version specified on the first line that `$VERSION` is mentioned.

Reply:
Just remove the `BEGIN` block, it's pointless.

Reply:
I can see reverting this change for now, but how can we proceed forward and enable strict and warnings in this module?
It seems that with such direction it would not be possible to enable strict and warnings.
I would expect that this syntax is common enough that modules like EUMM knows how to parse it.
If that's not the case, then I think we need to fix it upstream.

Reply:
I'm not sure why you think this matters to strict and warnings. The variable is not used again.

Reply:
Even if it was, you could just `our` it again in that scope, it would reference the same variable. Plus as @xenu mentioned the BEGIN block is likely unnecessary here.

Reply:
note that later we are doing  `BEGIN { XSLoader::load('Storable') };` so VERSION have to be set before

Reply:
@Grinnz good point, as it s not used, it s a problem for another day... going to revert that change and leave the our as it was earlier

Reply:
That does not contain any reference to the variable declared here, it would use the package variable directly.

Reply:
@Grinnz fixed by be2a5c504c374b2061df9780050f2d8223cf27ff
we would fix that problem another day

Reply:
the BEGIN block is required for the XSLoader... 
closing this conversation as this is now fixed, feel free to reopen it if needed